### DISCUSSION TITLE
PostActions (Run Script) - OS independent?

### DISCUSSION THREAD

**Discussion by umerkle** [dotnet/templating#7405] [Template authoring] [answered]
In the post action 'Run Script (3A7C4B45-1F5D-4A30-959A-51B88E82B5D2)', you can set an executable in the args object.
If you set it to, let's say 'setup.cmd', it will work on Windows and run setup.cmd. But if you use it in Linux, setup.cmd would not run, instead you want to run setup.sh or something like that.
Is it possible to have this os independent, or do you need special template for the different OSes? Or is it possible to select a postAction via condition depending on the OS?

**Comment by baronfel** [comment, marked as answer]
Hi @umerkle - for this I'd suggest that you use the `Condition` structure on the PostAction to filter down to the OS/platform that is correct for each RunScript PostAction. This would mean that you need two RunScript actions - one for Windows, one for non-Windows. You can see some examples of these in [this document](https://github.com/dotnet/templating/blob/main/docs/Post-Action-Registry.md#example-5).

Hope this helps!

**Reply by umerkle** [reply]
Hi @baronfel - cool! Thank you very much for the quick hint! That helped a lot...

**Reply by dblock247** [reply]
How are you determining the OS/Platform?
